app/screens/apps_form/apps_form_container.tsx

```diff
@@ -183,54 +185,33 @@ export default class AppsFormContainer extends PureComponent<Props, State> {
             },
             {details: message},
         );
-        const call = this.getCall();
-        if (!call) {
-            return makeErrorMsg(intl.formatMessage({id: 'apps.error.form.no_lookup_call', defaultMessage: 'performLookupCall props.call is not defined'}));
+        if (!field.lookup) {
+            return {error: makeCallErrorResponse(makeErrorMsg(intl.formatMessage({
+                id: 'apps.error.form.no_lookup',
+                defaultMessage: '`lookup` is not defined.',
+            })))};
         }
 
-        return this.props.actions.doAppCall({
-            ...call,
-            values,
-            selected_field: field.name,
-            query: userInput,
-        }, AppCallTypes.LOOKUP, intl);
-    }
-
-    getCall = (): AppCallRequest | null => {
-        const {form} = this.state;
-
-        const {call} = this.props;
-        if (!call) {
-            return null;
+        if (!this.props.context) {
+            return {error: makeCallErrorResponse('unreachable: empty context')};
         }
 
-        return {
-            ...call,
-            ...form?.call,
-            context: {
-                ...call.context,
-            },
-            values: {
-                ...call.values,
-            },
-        };
+        const creq = createCallRequest(field.lookup, this.props.context, {}, {
+            ...values,
+            [field.name]: userInput,
+        });
+        return this.props.actions.doAppLookup(creq, intl);
```

Comment:
@levb If I understand this correctly, `query: userInput` has changed to `[field.name]: userInput`.

I find this confusing, as `field.type` is a select element in this case, but its value is being "submitted" as a query string for the lookup. The type mismatch seems a little off.

What if we want to have a multiselect lookup field, and have the app check the currently selected values during the lookup? It seems there's no way to access that here.

Reply:
Discussed offline, decided to reinstate `CallRequest.LookupQuery` along with `LookupField`